examples/token-escrow/programs/token-escrow/src/escrow_with_compressed_pda/escrow.rs

```diff
@@ -0,0 +1,245 @@
+        bump,
+    )?;
+    Ok(())
+}
+
+fn cpi_compressed_pda_transfer<'info>(
+    ctx: &Context<'_, '_, '_, 'info, EscrowCompressedTokensWithCompressedPda<'info>>,
+    proof: Option<CompressedProof>,
+    new_address_params: NewAddressParamsPacked,
+    compressed_pda: CompressedAccount,
+    cpi_context: CompressedCpiContext,
+    bump: u8,
+) -> Result<()> {
+    let bump = &[bump];
+    let signer_bytes = ctx.accounts.signer.key.to_bytes();
+    let seeds = [b"escrow".as_slice(), signer_bytes.as_slice(), bump];
+    let inputs_struct = InstructionDataTransfer {
+        relay_fee: None,
+        input_compressed_accounts_with_merkle_context: Vec::new(),
+        output_compressed_accounts: vec![compressed_pda],
+        input_root_indices: Vec::new(),
+        output_state_merkle_tree_account_indices: vec![0],
+        proof,
+        new_address_params: vec![new_address_params],
+        compression_lamports: None,
+        is_compress: false,
+        signer_seeds: Some(seeds.iter().map(|x| x.to_vec()).collect::<Vec<Vec<u8>>>()),
+    };
+
+    let mut inputs = Vec::new();
+    InstructionDataTransfer::serialize(&inputs_struct, &mut inputs).unwrap();
+
+    let cpi_accounts = light_compressed_pda::cpi::accounts::TransferInstruction {
+        signer: ctx.accounts.token_owner_pda.to_account_info(),
+        registered_program_pda: ctx.accounts.registered_program_pda.to_account_info(),
+        noop_program: ctx.accounts.noop_program.to_account_info(),
+        account_compression_authority: ctx.accounts.account_compression_authority.to_account_info(),
+        account_compression_program: ctx.accounts.account_compression_program.to_account_info(),
+        invoking_program: Some(ctx.accounts.self_program.to_account_info()),
+        compressed_sol_pda: None,
+        compression_recipient: None,
+        system_program: None,
+        cpi_signature_account: Some(
+            ctx.remaining_accounts[cpi_context.cpi_signature_account_index as usize]
+                .to_account_info(),
+        ),
+    };
+    let seeds = [seeds.as_slice()];
+    let mut cpi_ctx = CpiContext::new_with_signer(
+        ctx.accounts.compressed_pda_program.to_account_info(),
+        cpi_accounts,
+        &seeds,
+    );
+
+    cpi_ctx.remaining_accounts = ctx.remaining_accounts.to_vec();
+
+    light_compressed_pda::cpi::execute_compressed_transaction(cpi_ctx, inputs, Some(cpi_context))?;
+    Ok(())
+}
+
+fn create_compressed_pda_data(
+    lock_up_time: u64,
+    ctx: &Context<'_, '_, '_, '_, EscrowCompressedTokensWithCompressedPda<'_>>,
+    new_address_params: &NewAddressParamsPacked,
+) -> Result<CompressedAccount> {
+    let current_slot = Clock::get()?.slot;
+    let timelock_compressed_pda = EscrowTimeLock {
+        slot: current_slot.checked_add(lock_up_time).unwrap(),
+    };
+    let compressed_account_data = CompressedAccountData {
+        discriminator: 1u64.to_le_bytes(),
+        data: timelock_compressed_pda.try_to_vec().unwrap(),
```

Comment:
Just making sure - we are not tight on the heap space here and it works reliably?

Reply:
it's just an example so I just focused on making it work, we can definitely improve it in another pr!